builtin/logical/database/path_config_connection.go

```diff
@@ -210,44 +204,40 @@ func (b *databaseBackend) connectionWriteHandler() framework.OperationFunc {
 		}
 
 		verifyConnection := data.Get("verify_connection").(bool)
-
 		allowedRoles := data.Get("allowed_roles").([]string)
+		rootRotationStatements := data.Get("root_rotation_statements").([]string)
 
 		// Remove these entries from the data before we store it keyed under
 		// ConnectionDetails.
 		delete(data.Raw, "name")
 		delete(data.Raw, "plugin_name")
 		delete(data.Raw, "allowed_roles")
 		delete(data.Raw, "verify_connection")
+		delete(data.Raw, "root_rotation_statements")
 
-		config := &DatabaseConfig{
-			ConnectionDetails: data.Raw,
-			PluginName:        pluginName,
-			AllowedRoles:      allowedRoles,
-		}
-
-		db, err := dbplugin.PluginFactory(ctx, config.PluginName, b.System(), b.logger)
+		// Create a database plugin and initialize it. This instance is not
+		// going to be used and is initialized just to ensure all parameters
+		// are valid and the connection is verified, if requested.
+		db, err := dbplugin.PluginFactory(ctx, pluginName, b.System(), b.logger)
 		if err != nil {
 			return logical.ErrorResponse(fmt.Sprintf("error creating database object: %s", err)), nil
 		}
-
-		err = db.Initialize(ctx, config.ConnectionDetails, verifyConnection)
+		connDetails, err := db.Initialize(ctx, data.Raw, verifyConnection)
+		db.Close()
```

Comment:
Why not keep this open if there were no errors?

Reply:
Since we no longer holding the lock here, I decided to not deal with the raciness here and let the first connection into the backend to get the connection and cache it in the backend.